src/Symfony/Component/Cache/Adapter/RedisTagAwareAdapter.php

```diff
@@ -161,46 +156,33 @@ protected function doDelete(array $ids, array $tagData = []): bool
      */
     protected function doInvalidate(array $tagIds): bool
     {
-        if (!$this->redisServerSupportSPOP()) {
-            return false;
-        }
-
-        // Pop all tag info at once to avoid race conditions
-        $tagIdSets = $this->pipeline(static function () use ($tagIds) {
-            foreach ($tagIds as $tagId) {
-                // Client: Predis or PHP Redis 3.1.3+ (https://github.com/phpredis/phpredis/commit/d2e203a6)
-                // Server: Redis 3.2 or higher (https://redis.io/commands/spop)
-                yield 'sPop' => [$tagId, self::POP_MAX_LIMIT];
+        // Rename all sets so we can safely avoid race conditions
+        $this->pipeline(static function () use ($tagIds) {
+            $uniqueToken = bin2hex(random_bytes(10));
+            foreach ($tagIds as $key => $tagId) {
+                $tagIds[$key] = $temporaryTagId = $tagId . ':' . $uniqueToken;
```

Comment:
For what purpose do you assign this value to `$tagIds[$key]`? As I see, you don't use this below.

```suggestion
               $temporaryTagId = $tagId . ':' . $uniqueToken;
```

In that case we don't need to assign `$key` too.

Reply:
Code can be more clear on this, but  `temporaryTagId`  is used in sscan loop below that needs some further work

Reply:
@andrerom afaik this changes applied only in current scope.
https://3v4l.org/WLgFN

Reply:
yes, I need to change it.

Note that this is a draft, not ready for review, I don't have things set up locally atm to test there so I pushed a draft to see what Travis said ;)